Mathlib/Algebra/GroupWithZero/NonZeroDivisors.lean

```diff
@@ -212,6 +212,16 @@ theorem map_mem_nonZeroDivisors [Nontrivial M] [NoZeroDivisors M'] [ZeroHomClass
     (hg : Function.Injective g) {x : M} (h : x ∈ M⁰) : g x ∈ M'⁰ := fun _ hz ↦
   eq_zero_of_ne_zero_of_mul_right_eq_zero (map_ne_zero_of_mem_nonZeroDivisors g hg h) hz
 
+theorem MulEquivClass.map_nonZeroDivisors {R S F : Type*} [MonoidWithZero R] [MonoidWithZero S]
+    [EquivLike F R S] [MulEquivClass F R S] (h : F) :
+    Submonoid.map h (nonZeroDivisors R) = nonZeroDivisors S := by
+  let h : R ≃* S := h
+  show Submonoid.map h.toMonoidHom _ = _
+  ext
+  simp_rw [Submonoid.map_equiv_eq_comap_symm, Submonoid.mem_comap, mem_nonZeroDivisors_iff,
+    ← h.symm.forall_congr_right, h.symm.coe_toMonoidHom, h.symm.toEquiv_eq_coe, h.symm.coe_toEquiv,
+    ← map_mul, map_eq_zero_iff _ h.symm.injective]
```

Comment:
probably
```suggestion
    ← map_mul, _root_.map_eq_zero_iff _ h.symm.injective]
```

Reply:
looks like to_additive alias malfunctioing in #18809 ...

Reply:
Oh, let's fix this first then.